modules/mvt/src/lib/geojson-vt/geojson-tiler.ts

```diff
@@ -0,0 +1,253 @@
+// loaders.gl, MIT license
+// Forked from https://github.com/mapbox/geojson-vt under compatible ISC license
+
+/* eslint-disable no-console, no-continue */
+
+import {convert} from './convert'; // GeoJSON conversion and preprocessing
+import {clip} from './clip'; // stripe clipping algorithm
+import {wrap} from './wrap'; // date line processing
+import {transformTile} from './transform'; // coordinate transformation
+import {createTile} from './tile'; // final simplified tile generation
+
+/** Options to configure tiling */
+export type GeoJSONTilerOptions = {
+  maxZoom?: number /** max zoom to preserve detail on */;
+  indexMaxZoom?: number /** max zoom in the tile index */;
+  indexMaxPoints?: number /** max number of points per tile in the tile index */;
+  tolerance?: number /** simplification tolerance (higher means simpler) */;
+  extent?: number /** tile extent */;
+  buffer?: number /** tile buffer on each side */;
+  lineMetrics?: boolean /** whether to calculate line metrics */;
+  promoteId?: string /** name of a feature property to be promoted to feature.id */;
+  generateId?: boolean /** whether to generate feature ids. Cannot be used with promoteId */;
+  debug?: number /** logging level (0, 1 or 2) */;
+};
+
+const defaultOptions: Required<GeoJSONTilerOptions> = {
+  maxZoom: 14, // max zoom to preserve detail on
+  indexMaxZoom: 5, // max zoom in the tile index
+  indexMaxPoints: 100000, // max number of points per tile in the tile index
+  tolerance: 3, // simplification tolerance (higher means simpler)
+  extent: 4096, // tile extent
+  buffer: 64, // tile buffer on each side
+  lineMetrics: false, // whether to calculate line metrics
+  // @ts-expect-error Ensures all these required params have defaults
+  promoteId: undefined, // name of a feature property to be promoted to feature.id
+  generateId: false, // whether to generate feature ids. Cannot be used with promoteId
+  debug: 0 // logging level (0, 1 or 2)
+};
+
+export class GeoJSONTiler {
+  options: Required<GeoJSONTilerOptions>;
+
+  // tiles and tileCoords are part of the public API
+  tiles: Record<string, any> = {};
+  tileCoords: any[] = [];
```

Comment:
I think this could be typed as
```suggestion
  tileCoords: {z: number; x: number; y: number;}[] = [];
```

Reply:
Good point. Will address in follow up.